src/Raven.Server/Json/BlittableNumber.cs

```diff
@@ -7,9 +7,9 @@ public static class BlittableNumber
     {
         public static NumberParseResult Parse(object value, out double doubleResult, out long longResult)
         {
-            if (value is long)
+            if (value is long || value is int)
             {
-                longResult = (long)value;
+                longResult = Convert.ToInt64(value);
```

Comment:
This is much more expensive, than a simple cast, though.
Note that I don't _think_ that we can have an int in blittable at all.